src/targets/services/aggregation.js

```diff
@@ -2,121 +2,71 @@ global.fetch = require('node-fetch').default
 global.btoa = require('btoa')
 
 import fs from 'fs'
-import CozyClient from 'cozy-client'
+import CozyClient, { Q } from 'cozy-client'
 import { Model } from './helpers'
 import dissecConfig from '../../../dissec.config.json'
 
 export const aggregation = async () => {
+  const client = CozyClient.fromEnv(process.env, {})
+
+  var originalConsoleLog = console.log
+  console.log = function () {
+    let args = []
+    args.push('[' + client.stackClient.uri.split('/')[2] + '] ')
+    for (var i = 0; i < arguments.length; i++) {
+      args.push(arguments[i])
+    }
+    originalConsoleLog.apply(console, args)
+  }
+
+  console.log('Aggregating', process.env['COZY_COUCH_DOC'])
+
   // Worker's arguments
   const {
-    docId,
-    sharecode,
-    uri,
+    _id,
+    dir_id: aggregationDirectory,
+    name,
+    metadata
+  } = JSON.parse(process.env['COZY_COUCH_DOC'] || {})
+
+  // This file is not a share uploaded by an aggregator for himself
+  if (!(metadata && metadata.dissec)) return
+
+  const {
+    executionId,
+    aggregatorId,
+    level,
     nbShares,
     parent,
     finalize,
-    level,
-    aggregatorId,
-    executionId,
     nbChild
-  } = JSON.parse(process.env['COZY_PAYLOAD'] || {})
-
-  const client = CozyClient.fromEnv(process.env, {})
+  } = metadata
 
-  // Download share using provided informations
-  const sharedClient = new CozyClient({
-    uri: uri,
-    token: sharecode,
-    schema: {
-      files: {
-        doctype: 'io.cozy.files',
-        relationships: {
-          old_versions: {
-            type: 'has-many',
-            doctype: 'io.cozy.files.versions'
-          }
-        }
-      }
-    },
-    store: false
-  })
-  const share = await sharedClient.stackClient.fetchJSON(
-    'GET',
-    `/files/download/${docId}`
-  )
-
-  // Storing shares as files to be shared
-  // Create or find a DISSEC directory
-  const baseFolder = 'DISSEC'
-  let dissecDirectory
-  try {
-    const { data } = await client.stackClient.fetchJSON(
-      'POST',
-      `/files/io.cozy.files.root-dir?Type=directory&Name=${baseFolder}`
-    )
-    dissecDirectory = data.id
-  } catch (e) {
-    const { included } = await client.stackClient.fetchJSON(
-      'GET',
-      '/files/io.cozy.files.root-dir'
-    )
-    dissecDirectory = included.filter(
-      dir => dir.attributes.name === baseFolder
-    )[0].id
-  }
+  // TODO: Remove hierarchy and base only on metadata and id
+  // Count files in the aggregation folder
+  const { data: unfilteredFiles } = await client.query(Q('io.cozy.files').where({
+    dir_id: aggregationDirectory
+  }))
+  const receivedShares = unfilteredFiles.filter(file => file.attributes.metadata && (file.attributes.metadata.level === level))
```

Comment:
We should eventually add travis, so he can complain about linting issues like this one (line too long) :)